src/dal_select2/views.py

```diff
@@ -44,3 +45,47 @@ def render_to_response(self, context):
 
 class Select2QuerySetView(Select2ViewMixin, BaseQuerySetView):
     """List options for a Select2 widget."""
+
+class Select2ListView(ViewMixin, View):
+    """Autocomplete from a list of items rather than a QuerySet.
+    get_list() should return a list of strings that may be autocopleted.
+    """
+
+    create_field = False
+    """Set create_field to True to allow creation of new items."""
+
+    def get_list(self):
+        """"Return the list strings from which to autocomplete."""
+        return []
+
+    def get(self, request, *args, **kwargs):
+        results = self.get_list()
+        create_option = []
+        if self.q:
+            results = [ x for x in results if self.q in x ]
+            if self.create_field:
+                create_option = [{
+                        'id': self.q,
+                        'text': 'Create "%s"' % self.q,
+                        'create_id': True
+                    }]
+        return http.HttpResponse(json.dumps({
+            'results': [dict(id=x, text=x) for x in results] + create_option
+        }))
+
+    def create(self, text):
+        """"Create a new list item for text.  None can be returned if the
+        item may not be created for some reason."""
+        return text
```

Comment:
On Sun, Mar 27, 2016 at 10:03 PM, James Pic <notifications@github.com>
wrote:

> In src/dal_select2/views.py
> <https://github.com/yourlabs/django-autocomplete-light/pull/644#discussion_r57543419>
> :
>
> > +        if self.q:
> > +            results = [ x for x in results if self.q in x ]
> > +            if self.create_field:
> > +                create_option = [{
> > +                        'id': self.q,
> > +                        'text': 'Create "%s"' % self.q,
> > +                        'create_id': True
> > +                    }]
> > +        return http.HttpResponse(json.dumps({
> > +            'results': [dict(id=x, text=x) for x in results] + create_option
> > +        }))
> > +
> > +    def create(self, text):
> > +        """"Create a new list item for text.  None can be returned if the
> > +        item may not be created for some reason."""
> > +        return text
>
> Shouldn't we raise NotImplemented or something like that here ?
>
I don't feel very strongly about it.  I thought that if you set
create_field to True, you'd expect that any value would be accepted for
creation by default.  What's your thought process?

  --david​
-- 
Love Linux kernel hacking?  Know Python?  We are hiring!  Contact us at
jobs@egauge.net